datafusion/sqllogictest/test_files/map.slt

```diff
@@ -44,6 +44,7 @@ DELETE 24
 query T
 SELECT strings['not_found'] FROM data LIMIT 1;
 ----
+NULL
```

Comment:
I'm not familiar with Map. Why should we returns null here?

Reply:
It will throws the invalidation error i added in this PR. I think the correct behavior is to return null for every input rows. I took a look at duckdb and spark's behavior

```
import org.apache.spark.sql.functions._
import org.apache.spark.sql.SparkSession

val spark = SparkSession.builder().appName("Spark SQL Map Example").getOrCreate()
import spark.implicits._

val data = Seq(
  ("Alice", Map("age" -> "25", "email" -> "alice@example.com")),
  ("Bob", Map("age" -> "30", "email" -> "bob@example.com")),
  ("Carol", Map("age" -> "35", "email" -> "carol@example.com"))
)
val df = data.toDF("name", "attributes")
val result = df.select($"name", $"attributes.email".as("email"),$"attributes.notfound".as("should_be_null"))
// Show the DataFrame
result.show(false)

+-----+-----------------+--------------+
|name |email            |should_be_null|
+-----+-----------------+--------------+
|Alice|alice@example.com|NULL          |
|Bob  |bob@example.com  |NULL          |
|Carol|carol@example.com|NULL          |
+-----+-----------------+--------------+
```

And also, a similar implementation in Datafusion is array_element also return null if the index goes out of range

Reply:
Here is how it works on main:

```sql
> select strings['not_found'] from 'datafusion/core/tests/data/parquet_map.parquet';
0 row(s) fetched.
Elapsed 0.006 seconds.

```

Here is how it works on this PR (aka has a single row for each input row) 
```sql
DataFusion CLI v37.1.0
> select strings['not_found'] from '../datafusion/core/tests/data/parquet_map.parquet';
+----------------------------------------------------------------------+
| ../datafusion/core/tests/data/parquet_map.parquet.strings[not_found] |
+----------------------------------------------------------------------+
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
|                                                                      |
| .                                                                    |
| .                                                                    |
| .                                                                    |
+----------------------------------------------------------------------+
209 row(s) fetched. (First 40 displayed. Use --maxrows to adjust)
Elapsed 0.033 seconds.
```